crossplane/objects.py

```diff
@@ -33,23 +41,77 @@ def get(self, directive):
         """
         return []
 
-    def to_dict(self):
+    def to_crossplane(self, include_meta=True):
         result = {}
 
+        # exclude slots unique to object layer
         for slot in self.__slots__:
-            if slot not in ('parent',):
+            if slot not in self.__meta_slots__:
                 result[slot] = getattr(self, slot)
 
         return result
 
+    def hexdigest(self):
+        crossplane_dict = self.to_crossplane(include_meta=False)
+
+        if PYTHON3:
+            return hashlib.sha256(
+                str(crossplane_dict).encode('utf-8')
+            ).hexdigest()
+        else:
+            return hashlib.sha256(
+                unicode(crossplane_dict).encode('utf-8')
+            ).hexdigest()
```

Comment:
you should add `unicode` to `compat.py` like this:
```python
if PY2:
    # ...
    unicode = unicode
else:
    # ...
    unicode = str
```
and then import it at the top like:
```python
from .compat import unicode
```